### ISSUE TITLE
Ingestion Status broken when multiple ingestions are started for a collection in LOCAL  

### ISSUE THREAD

**Issue by kunwar31** [truefoundry/cognita#329]
Steps to reproduce:
1. Create a collection
2. Link 2 data sources to it.

Excepted Behaviour:

- when one of the data sources has completed ingestion, its status changes to COMPLETED

Observed Behaviour:

- Status changes to completed only when both ingestions are completed
- in the attached screenshots, one of the data source has completed ingestion (see unstructured_api CPU usage, it was 200% when both were running) , but both have a status of DATA_INGESTION_STARTED. 
![image](https://github.com/user-attachments/assets/0cca176c-a63b-4572-9d1f-a2bf4fca5c04)
![image](https://github.com/user-attachments/assets/d527b540-24e5-4f75-8582-b83e7c12f97c)

- when both are completed, the status for both of them is set to COMPLETED
![image](https://github.com/user-attachments/assets/afbe9c11-4b47-45b9-8083-2b67ecf38487)

**Comment by S1LV3RJ1NX** [comment]
Yes, actually same error occurs in deployed version as well, we shall check on this.
